administration/src/application/components/forms/RegionForm.tsx

```diff
@@ -0,0 +1,69 @@
+import { Region, ShortTextInput, useGetRegionByPostalCodeQuery } from '../../../generated/graphql'
+import { useUpdateStateCallback } from '../../useUpdateStateCallback'
+import { Form } from '../../FormType'
+import {
+  CompoundState,
+  createCompoundGetArrayBufferKeys,
+  createCompoundValidate,
+  createCompoundInitialState,
+} from '../../compoundFormUtils'
+import SelectForm, { SelectItem } from '../primitive-inputs/SelectForm'
+import { ProjectConfigContext } from '../../../project-configs/ProjectConfigContext'
+import { useContext } from 'react'
+import { useSnackbar } from 'notistack'
+
+const SubForms = {
+  region: SelectForm,
+}
+
+type State = CompoundState<typeof SubForms>
+type ValidatedInput = { region: ShortTextInput }
+type Options = {}
+type AdditionalProps = { regionData: Region[]; postalCode: string }
+
+const getOptionsLabel = (prefix: string, name: string) => `${name} (${prefix})`
+
+export const getOptions = (regions: Region[]): SelectItem[] =>
+  regions.map(region => {
+    return { label: getOptionsLabel(region.prefix, region.name), value: region.id.toString() }
+  })
+
+const RegionForm: Form<State, Options, ValidatedInput, AdditionalProps> = {
+  initialState: { ...createCompoundInitialState(SubForms) },
+  getArrayBufferKeys: createCompoundGetArrayBufferKeys(SubForms),
+  validate: createCompoundValidate(SubForms, { region: { items: [] } }),
+  Component: ({ state, setState, regionData, postalCode }) => {
+    const projectId = useContext(ProjectConfigContext).projectId
+    const { enqueueSnackbar } = useSnackbar()
+    const {} = useGetRegionByPostalCodeQuery({
+      onError: () => {
```

Comment:
I think its nice for the user to have a indicator if region could be auto selected or not

Reply:
I think it is a bit confusing that the snackbars appear with text that I don't yet understand. (I don't really know what it's talking about, when I am still entering my address).
I think it would be better to have explanations of what happened directly at the select form (I would suggest removing the snackbars in favor of the suggestions below). [Also note, that snackbars are not really accessible -- at least not in our current implementation I think. We should avoid using them too often]

Reply:
implemented the alerts